File: CS/ConnectToRouteService/Form1.cs
using DevExpress.XtraMap;
using System.Collections.Generic;
using System.Drawing;
using System.Windows.Forms;

namespace ConnectToRouteService {
    public partial class Form1 : DevExpress.XtraEditors.XtraForm {
        const string yourBingKey = "YOUR BING KEY";
        BingRouteDataProvider routeProvider;

        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, System.EventArgs e) {
            imageProvider.BingKey = yourBingKey;

            routeProvider = new BingRouteDataProvider { BingKey = yourBingKey };
            informationLayer.DataProvider = routeProvider;
            #region #Waypoints
            // Create three waypoints and add them to a route waypoints list. 
            List<RouteWaypoint> waypoints = new List<RouteWaypoint> {
                new RouteWaypoint("New York", new GeoPoint(41.145556, -73.995)),
                new RouteWaypoint("Oklahoma", new GeoPoint(36.131389, -95.937222)),
                new RouteWaypoint("Las Vegas", new GeoPoint(36.175, -115.136389))
            };
            #endregion #Waypoints
            // Call the BingRouteDataProvider.CalculateRoute method.
            splashScreenManager.ShowWaitForm();
            routeProvider.CalculateRoute(waypoints);

            // Handle the BingRouteDataProvider.LayerItemsGenerating event.
            routeProvider.LayerItemsGenerating += routeLayerItemsGenerating;
        }

        private void routeLayerItemsGenerating(object sender, LayerItemsGeneratingEventArgs e) {
            if(e.Cancelled || (e.Error != null)) return;
            
            //char pushpinMarker = 'A';
            foreach(MapItem item in e.Items) {
                //MapPushpin pushpin = item as MapPushpin;
                //if(pushpin != null) {
                //    pushpin.Text = pushpinMarker++.ToString();
                //}
                MapPolyline polyline = item as MapPolyline;
                if(polyline != null) {
                    polyline.Stroke = Color.FromArgb(0xFF, 0x00, 0x72, 0xC6);
                    polyline.StrokeWidth = 4;
                }
            }
            splashScreenManager.CloseWaitForm();
            mapControl.ZoomToFit(e.Items, 0.4);
        }
    }
}

File: VB/ConnectToRouteService/Form1.vb
Imports DevExpress.XtraMap
Imports System.Collections.Generic
Imports System.Drawing

Namespace ConnectToRouteService

    Public Partial Class Form1
        Inherits DevExpress.XtraEditors.XtraForm

        Const yourBingKey As String = "YOUR BING KEY"

        Private routeProvider As BingRouteDataProvider

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As System.EventArgs)
            imageProvider.BingKey = yourBingKey
            routeProvider = New BingRouteDataProvider With {.BingKey = yourBingKey}
            informationLayer.DataProvider = routeProvider
#Region "#Waypoints"
            ' Create three waypoints and add them to a route waypoints list. 
            Dim waypoints As List(Of RouteWaypoint) = New List(Of RouteWaypoint) From {New RouteWaypoint("New York", New GeoPoint(41.145556, -73.995)), New RouteWaypoint("Oklahoma", New GeoPoint(36.131389, -95.937222)), New RouteWaypoint("Las Vegas", New GeoPoint(36.175, -115.136389))}
#End Region  ' #Waypoints
            ' Call the BingRouteDataProvider.CalculateRoute method.
            splashScreenManager.ShowWaitForm()
            routeProvider.CalculateRoute(waypoints)
            ' Handle the BingRouteDataProvider.LayerItemsGenerating event.
            AddHandler routeProvider.LayerItemsGenerating, AddressOf routeLayerItemsGenerating
        End Sub

        Private Sub routeLayerItemsGenerating(ByVal sender As Object, ByVal e As LayerItemsGeneratingEventArgs)
            If e.Cancelled OrElse e.Error IsNot Nothing Then Return
            'char pushpinMarker = 'A';
            For Each item As MapItem In e.Items
                'MapPushpin pushpin = item as MapPushpin;
                'if(pushpin != null) {
                '    pushpin.Text = pushpinMarker++.ToString();
                '}
                Dim polyline As MapPolyline = TryCast(item, MapPolyline)
                If polyline IsNot Nothing Then
                    polyline.Stroke = Color.FromArgb(&HFF, &H00, &H72, &HC6)
                    polyline.StrokeWidth = 4
                End If
            Next

            splashScreenManager.CloseWaitForm()
            mapControl.ZoomToFit(e.Items, 0.4)
        End Sub
    End Class
End Namespace
